crates/torii/graphql/src/object/model_data.rs

```diff
@@ -111,10 +108,106 @@ impl ObjectTrait for ModelStateObject {
         // Add relay connection fields (first, last, before, after, where)
         field = connection_arguments(field);
         field = where_argument(field, self.type_name());
-        field = order_argument(field, self.type_name());
+        // field = order_argument(field, self.type_name());
 
         Some(field)
     }
+
+    fn objects(&self) -> Vec<Object> {
+        let mut path_array = vec![self.type_name().to_string()];
+        let mut objects = data_objects(self.type_name(), self.type_mapping(), &mut path_array);
+
+        // root object requires entity_field association
+        let mut root = objects.pop().unwrap();
+        root = root.field(entity_field());
+
+        objects.push(root);
+        objects
+    }
+}
+
+fn data_objects(
+    type_name: &str,
+    type_mapping: &TypeMapping,
+    path_array: &mut Vec<String>,
+) -> Vec<Object> {
+    let mut objects = Vec::<Object>::new();
+
+    for (_, type_data) in type_mapping {
+        if let TypeData::Nested((nested_type, nested_mapping)) = type_data {
+            path_array.push(nested_type.to_string());
+            objects.extend(data_objects(
+                &nested_type.to_string(),
+                nested_mapping,
+                &mut path_array.clone(),
+            ));
+        }
+    }
+
+    objects.push(object(type_name, type_mapping, path_array));
+    objects
+}
+
+pub fn object(type_name: &str, type_mapping: &TypeMapping, path_array: &[String]) -> Object {
+    let mut object = Object::new(type_name);
+
+    for (field_name, type_data) in type_mapping.clone() {
+        let table_name = path_array.join("$");
+
+        let field = Field::new(field_name.to_string(), type_data.type_ref(), move |ctx| {
+            let field_name = field_name.clone();
+            let type_data = type_data.clone();
+            let table_name = table_name.clone();
+
+            // Field resolver for nested types
+            if let TypeData::Nested((_, nested_mapping)) = type_data {
+                return FieldFuture::new(async move {
+                    match ctx.parent_value.try_to_value()? {
+                        Value::Object(indexmap) => {
+                            let mut conn = ctx.data::<Pool<Sqlite>>()?.acquire().await?;
+                            let entity_id = extract::<String>(indexmap, "entity_id")?;
+
+                            // TODO: remove subqueries and use JOIN in parent query
```

Comment:
ah yeah this will be a big optimization at some point. we should track it with an issue